types/pdf-thumbnail/index.d.ts

```diff
@@ -0,0 +1,37 @@
+// Type definitions for pdf-thumbnail 1.0
+// Project: https://github.com/nicoFuccella/pdf-thumbnail#readme
+// Definitions by: Kirill Koshil <https://github.com/koshilki>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+export type Operations = 'compress' | 'crop' | 'resize';
+
+export interface CompressParams {
+    type?: string;
+    quality?: number;
+}
+
+export interface CropParams {
+    width: number;
+    height: number;
+    x: number;
+    y: number;
+    ratio?: boolean;
+}
+
+export interface ResizeParams {
+    width?: number;
+    height?: number;
+}
+
+export interface OperationsParams {
+    compress: CompressParams;
+    crop: CropParams;
+    resize: ResizeParams;
+}
+
+export type PDFThumbnailOptions = {
+    [op in Operations]?: OperationsParams[op];
+};
+
+export default function pdf(body: NodeJS.ReadableStream | Buffer | string, options?: PDFThumbnailOptions): Promise<NodeJS.ReadableStream>;
```

Comment:
pdf-thumbnail is not an ES module, so there is no default export. This whole module should be written as

```ts
declare namespace pdf {
  export type Operations = 'compress' | 'crop' | 'resize';

  export interface CompressParams {
    type?: string;
   quality?: number;
  }

  // ...
}

declare function pdf(body: NodeJS.ReadableStream | Buffer | string, options?: PDFThumbnailOptions): Promise<NodeJS.ReadableStream>;

export = pdf;
```

Reply:
Thank you for the review, good point. Fixed